### ISSUE TITLE
The latest version of NGen.cpp causes a build error on GCC8 (Linux)

### ISSUE THREAD

**Issue by stcui007** [NOAA-OWP/ngen#635]
In build ngen with the latest master, it produces an error on GCC8 due to a missing "std" on line 298.

## Current behavior
```[ 97%] Building CXX object CMakeFiles/ngen.dir/src/NGen.cpp.o
/home/shengting.cui/ngen_9.11.23/ngen/src/NGen.cpp: In function ‘int main(int, char**)’:
/home/shengting.cui/ngen_9.11.23/ngen/src/NGen.cpp:298:16: error: ‘make_unique’ was not declared in this scope
       router = make_unique<routing_py_adapter::Routing_Py_Adapter>(t_route_config_file_with_path);
                ^~~~~~~~~~~
/home/shengting.cui/ngen_9.11.23/ngen/src/NGen.cpp:298:16: note: suggested alternative:
In file included from /opt/rh/devtoolset-8/root/usr/include/c++/8/memory:80,
                 from /home/shengting.cui/ngen_9.11.23/ngen/include/realizations/catchment/Formulation_Manager.hpp:4,
                 from /home/shengting.cui/ngen_9.11.23/ngen/src/NGen.cpp:8:
/opt/rh/devtoolset-8/root/usr/include/c++/8/bits/unique_ptr.h:842:5: note:   ‘std::make_unique’
     make_unique(_Args&&...) = delete;```

## Expected behavior
No error output

at
1.


## Screenshots

**Comment by program--** [comment]
For reference: https://github.com/NOAA-OWP/ngen/blob/b749e0fd0d3766349f368d2a5eaa31003601b5fe/src/NGen.cpp#L298

**Comment by stcui007** [comment]
It appears this occurs only when ```NGEN_ROUTING_ACTIVE``` is turned on.

**Comment by mattw-nws** [comment]
Curiously, I wasn't getting this on clang with NGEN_ROUTING_ACTIVE enabled... but after rebasing the branch I was on I get it now. In any case, yes this should have `std::` stuck on the front of it (just not sure why _now_).

Related, @PhilMiller did you spare your wrath from https://github.com/NOAA-OWP/ngen/blob/b749e0fd0d3766349f368d2a5eaa31003601b5fe/test/realizations/catchments/Bmi_Testing_Util.hpp#L59 and https://github.com/NOAA-OWP/ngen/blob/b749e0fd0d3766349f368d2a5eaa31003601b5fe/test/realizations/catchments/Bmi_C_Pet_IT.cpp#L23 in #603 on purpose? Those are sticking around after the zombie apocalypse. Whether Bmi_C_Pet_IT.cpp belongs where it is is debatable, but at least all the code it's testing is still in use.

**Comment by PhilMiller** [comment]
Fixed by #636

**Comment by PhilMiller** [comment]
> Related, @PhilMiller did you spare your wrath from
> 
> https://github.com/NOAA-OWP/ngen/blob/b749e0fd0d3766349f368d2a5eaa31003601b5fe/test/realizations/catchments/Bmi_Testing_Util.hpp#L59
> and
> 
> https://github.com/NOAA-OWP/ngen/blob/b749e0fd0d3766349f368d2a5eaa31003601b5fe/test/realizations/catchments/Bmi_C_Pet_IT.cpp#L23
> in #603 on purpose? Those are sticking around after the zombie apocalypse. Whether Bmi_C_Pet_IT.cpp belongs where it is is debatable, but at least all the code it's testing is still in use.

Documenting from Slack:
I left some of the test code alone as less impactful for the substantial effort. The test .cpp file won't hurt anything else. I neglected that there's a .hpp file, which maybe should be cleaned up.
